internal/pkg/build/remotebuilder/remotebuilder.go

```
@@ -84,159 +80,87 @@ func (rb *RemoteBuilder) Build(ctx context.Context) (err error) {
 		fmt.Printf("Alternatively, you can access it from a browser at:\n\t%v/library/%v\n", CloudURI, libraryRefRaw)
+		return nil
 	}
 
-	// If we're doing an attached build, stream output and then download the resulting file
-	if !rb.IsDetached {
-		err = rb.streamOutput(ctx, rd.WSURL)
-		if err != nil {
-			err = errors.Wrap(err, "failed to stream output from remote build service")
-			sylog.Warningf("%v", err)
-			return err
-		}
-
-		// Get build status
-		rd, err = rb.doStatusRequest(ctx, rd.ID)
-		if err != nil {
-			err = errors.Wrap(err, "failed to get status from remote build service")
-			sylog.Warningf("%v", err)
-			return err
-		}
-
-		// Do not try to download image if not complete or image size is 0
-		if !rd.IsComplete {
-			return errors.New("build has not completed")
-		}
-		if rd.ImageSize <= 0 {
-			return errors.New("build image size <= 0")
-		}
-
-		// If image destination is local file, pull image.
-		if !strings.HasPrefix(rb.ImagePath, "library://") {
-			err = client.DownloadImage(rb.ImagePath, rd.LibraryRef, rd.LibraryURL, rb.Force, rb.AuthToken)
-			if err != nil {
-				err = errors.Wrap(err, "failed to pull image file")
-				sylog.Warningf("%v", err)
-				return err
-			}
-		}
-	}
-
-	return nil
-}
-
-// streamOutput attaches via websocket and streams output to the console
-func (rb *RemoteBuilder) streamOutput(ctx context.Context, url string) (err error) {
-	h := http.Header{}
-	rb.setAuthHeader(h)
-	h.Set("User-Agent", useragent.Value())
-
-	c, resp, err := websocket.DefaultDialer.Dial(url, h)
+	// We're doing an attached build, stream output and then download the resulting file
+	var outputLogger stdoutLogger
+	err = rb.BuildClient.GetOutput(ctx, bi.ID, outputLogger)
 	if err != nil {
-		sylog.Debugf("websocket dial err - %s, partial response: %+v", err, resp)
+		err = errors.Wrap(err, "failed to stream output from remote build service")
+		sylog.Warningf("%v", err)
```

Here too?

---

Trying to keep current pattern consistent. Fine with changing the behavior, just want it to be consistent.

---

Yeah, I dont think its necessary to print the err, then return it and print it again. It should just return the err in all cases, imo.